Rdmp.Core/DataFlowPipeline/Requirements/DataFlowPipelineContext.cs

```diff
@@ -216,47 +216,54 @@ public void PreInitialize(IDataLoadEventListener listener, IDataFlowSource<T> co
                
         private void PreInitializeComponentWithAllObjects(IDataLoadEventListener listener, object component, params object[] parameters)
         {
-            Dictionary<object, Dictionary<MethodInfo, object>> initializedComponents = new Dictionary<object, Dictionary<MethodInfo, object>>();
-
             //these are all the interfaces like IPipelineRequirement<TableInfo> etc
             var requirements = component.GetType().GetInterfaces().Where(i =>
                 i.IsGenericType && 
                 i.GetGenericTypeDefinition() 
                 == typeof(IPipelineRequirement<>)).ToArray();
 
+            Satisfy(requirements, false, listener, component,parameters);
+
+            // satisfy optional requirements
+            //these are all the interfaces like IPipelineOptionalRequirement<TableInfo> etc
+            var optionals = component.GetType().GetInterfaces().Where(i =>
+                i.IsGenericType &&
+                i.GetGenericTypeDefinition()
+                == typeof(IPipelineOptionalRequirement<>)).ToArray();
+
+            Satisfy(optionals, true, listener, component, parameters);
+        }
+
+        private void Satisfy(Type[] requirements, bool isOptional, IDataLoadEventListener listener, object component, params object[] parameters)
+        {
+            Dictionary<object, Dictionary<MethodInfo, object>> initializedComponents = new Dictionary<object, Dictionary<MethodInfo, object>>();
+
             //these are the T tokens in the above interfaces
             var typesRequired = requirements.Select(i => i.GenericTypeArguments[0]).ToList();
-
-            // Check if we have some PreInitialize functions for which there are no IPipelineRequirements, most likely an oversight one way or the other
-            var preInitializeFunctions = component.GetType().GetMethods().Where(mi => mi.Name == "PreInitialize");
```

Comment:
I've removed this check.  It looks for classes with methods called `PreInitialize` and looks to check for a corresponding interface declaration of `IPipelineRequirement<T>`.  This is not really our responsibility to check that the programmer is coding correctly.  

Even if it were, theres a number of problems with it including declaring an empty method (i.e. `PreInitialize()`).  The fact that it can now come from either `IPipelineOptionalRequirement` or `IPipelineRequirement`.  I think we are better off without it.